#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Parquetry.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.03285859866666669 0.04486148645330856 0.9984526826679683 1.2659053869838208
  position -0.33274384960480635 -0.9438037258987457 0.41752288551475913
}
Robot {
  translation 0 0 0.169
  rotation 0 1 0 0
  children [
    Solid {
      translation 0 0 0.025
      name "fuselage"
      boundingObject Capsule {
        height 0.135
        radius 0.05
        subdivision 16
      }
    }
    Propeller {
      shaftAxis 0.173648 0 0.984808
      centerOfThrust -0.1275 0.1275 0.105
      thrustConstants 1.437e-05 0
      torqueConstants 1.9e-06 0
      fastHelixThreshold 700
      device RotationalMotor {
        name "motor_xmyp"
        maxVelocity 683.82
        maxTorque 600
      }
      slowHelix Solid {
        translation -0.1275 0.1275 0.105
        rotation 0.17364796311995495 0 0.9848077908426043 5.730017332084662
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              metalness 0
            }
            geometry Mesh {
              url [
                "prop_scaled.stl"
              ]
            }
          }
        ]
        boundingObject Box {
          size 0.02 0.1 0.01
        }
        physics Physics {
          mass 0.001
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            1 1 1
            0 0 0
          ]
        }
      }
    }
    Propeller {
      shaftAxis 0.173648 0 0.984808
      centerOfThrust -0.1275 -0.1275 0.105
      thrustConstants -1.437e-05 0
      torqueConstants 1.9e-06 0
      fastHelixThreshold 700
      device RotationalMotor {
        name "motor_xmym"
        maxVelocity 683.82
        maxTorque 600
      }
      slowHelix Solid {
        translation -0.1275 -0.1275 0.105
        rotation 0.1736479631199549 0 0.9848077908426043 0.28851691180781225
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              metalness 0
            }
            geometry Mesh {
              url [
                "prop_scaled.stl"
              ]
            }
          }
        ]
        boundingObject Box {
          size 0.02 0.1 0.01
        }
        physics Physics {
          mass 0.001
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            1 1 1
            0 0 0
          ]
        }
      }
    }
    Propeller {
      shaftAxis -0.173648 0 0.984808
      centerOfThrust 0.1275 0.1275 0.105
      thrustConstants -1.437e-05 0
      torqueConstants 1.9e-06 0
      fastHelixThreshold 700
      device RotationalMotor {
        name "motor_xpyp"
        maxVelocity 683.82
        maxTorque 600
      }
      slowHelix Solid {
        translation 0.1275 0.1275 0.105
        rotation -0.1736479631199549 0 0.9848077908426043 0.28851691180781225
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              metalness 0
            }
            geometry Mesh {
              url [
                "prop_scaled.stl"
              ]
            }
          }
        ]
        boundingObject Box {
          size 0.02 0.1 0.01
        }
        physics Physics {
          mass 0.001
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            1 1 1
            0 0 0
          ]
        }
      }
    }
    Propeller {
      shaftAxis -0.173648 0 0.984808
      centerOfThrust 0.1275 -0.1275 0.105
      thrustConstants 1.437e-05 0
      torqueConstants 1.9e-06 0
      fastHelixThreshold 700
      device RotationalMotor {
        name "motor_xpym"
        maxVelocity 683.82
        maxTorque 600
      }
      slowHelix Solid {
        translation 0.1275 -0.1275 0.105
        rotation 0.1736479631199549 -4.010978321950238e-17 -0.9848077908426044 4.754700820015718
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              metalness 0
            }
            geometry Mesh {
              url [
                "prop_scaled.stl"
              ]
            }
          }
        ]
        boundingObject Box {
          size 0.02 0.1 0.01
        }
        physics Physics {
          mass 0.001
          centerOfMass [
            0 0 0
          ]
          inertiaMatrix [
            1 1 1
            0 0 0
          ]
        }
      }
    }
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Mesh {
        url [
          "drone_solid_scaled.stl"
        ]
      }
      castShadows FALSE
    }
    Accelerometer {
    }
    InertialUnit {
      name "imu"
    }
    Gyro {
    }
    GPS {
    }
  ]
  name "tester_drone"
  boundingObject Box {
    size 0.5 0.34 0.34
  }
  physics Physics {
    mass 1.8
    centerOfMass [
      0 0 0.015
    ]
    inertiaMatrix [
      0.014 0.012 0.02
      0 0 0
    ]
  }
  controller "quadcopter_test1"
}
TexturedBackground {
  texture "stadium"
}
RectangleArena {
  floorSize 50 50
  floorAppearance Parquetry {
    type "light strip"
  }
  wallHeight 5
  wallAppearance PBRAppearance {
    baseColor 0.482353 0.482353 0.482353
  }
}
